rancher/rancher
# Update rancher/webhook

## Commit 1:  Allow marshalling map[any]any for chart values

Without this change, if the rancher-config config map has nested objects
as chart values, e.g.:

```
data:
  rancher-webhook: |-
    auth:
      clientCA: foo
```

The resulting marshalled YAML will be a map[any]any, and the
unmarshalled object cannot be decoded by the catalog installer. This
change uses an alternative library so that nested values can be encoded.

## Commit 2 Update rancher/webhook

Depends on https://github.com/rancher/webhook/pull/273 and upcoming charts change

## Problem

The webhook has new optional chart values that need to be configurable by the user. These values are nested, and the current implementation for configuring webhook chart values does not support nested objects.
 
## Solution

Use an alternate library to encode charts values and bump the webhook version.
 
## Testing


## Engineering Testing
### Manual Testing

- built a custom charts repo with the updated webhook change and configured rancher to use it instead of git.rancherio
- see linked issue for full configuration

### Automated Testing

* Test types added/modified:
    * Unit

Summary: _TODO_

## QA Testing Considerations

 
### Regressions Considerations


With no new configmap values set, webhook should continue to deploy on rancher server and agents and work as normal.

Existing / newly added automated tests that provide evidence there are no regressions:
* catalogv2 unit tests
